internal/resource/config/datastore/custom_data_store.go

```diff
@@ -57,7 +59,20 @@ func toSchemaCustomDataStore() schema.SingleNestedAttribute {
 		"plugin_descriptor_ref": schema.SingleNestedAttribute{
 			Required:    true,
 			Description: "Reference to the plugin descriptor for this instance. The plugin descriptor cannot be modified once the instance is created. Note: Ignored when specifying a connection's adapter override.",
-			Attributes:  resourcelink.ToSchema(),
+			Attributes: map[string]schema.Attribute{
+				"id": schema.StringAttribute{
+					Description: "The ID of the resource.",
+					Required:    true,
+				},
+				"location": schema.StringAttribute{
+					Description: "A read-only URL that references the resource. If the resource is not currently URL-accessible, this property will be null.",
+					Computed:    true,
+					Optional:    false,
+					PlanModifiers: []planmodifier.String{
+						stringplanmodifier.UseStateForUnknown(),
```

Comment:
Why is this one the exception? What if someone wants to change the plugin_descriptor_ref here?

Reply:
Due to inconsistent plan when set is in the same schema.